spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-config/src/main/java/com/alibaba/cloud/nacos/configdata/NacosConfigDataLocationResolver.java

```diff
@@ -0,0 +1,295 @@
+		registerOrUpdateIndexes(properties, bootstrapContext);
+
+		// Retain the processing logic of the old version.
+		// Make sure to upgrade smoothly.
+		return loadConfigDataResources(resolverContext, location, profiles, properties);
+	}
+
+	private List<NacosConfigDataResource> loadConfigDataResources(
+			ConfigDataLocationResolverContext resolverContext,
+			ConfigDataLocation location, Profiles profiles,
+			NacosConfigProperties properties) {
+		List<NacosConfigDataResource> result = new ArrayList<>();
+		String uris = getUri(location, properties);
+
+		if (StringUtils.isNotBlank(dataIdFor(uris))) {
+			// Can be considered as an extension-config.
+			NacosConfigDataResource resource = loadResource(properties,
+					location.isOptional(), profiles, uris, dataIdFor(uris));
+			result.add(resource);
+		}
+		else {
+			String prefix = dataIdPrefixFor(resolverContext.getBinder(), properties);
+			NacosConfigDataResource resource = loadResource(properties,
+					location.isOptional(), profiles, uris, prefix);
+			result.add(resource);
+
+			resource = loadResource(properties, location.isOptional(), profiles, uris,
+					prefix + DOT + properties.getFileExtension());
+			result.add(resource);
+
+			for (String profile : profiles.getActive()) {
+				String dataId = prefix + HYPHEN + profile + DOT
+						+ properties.getFileExtension();
+				resource = loadResource(properties, location.isOptional(), profiles, uris,
+						dataId);
+				result.add(resource);
+			}
+		}
+		return result;
+	}
+
+	private NacosConfigDataResource loadResource(NacosConfigProperties properties,
+			boolean optional, Profiles profiles, String uris, String dataId) {
+		return new NacosConfigDataResource(properties, optional, profiles,
+				log,
+				new NacosItemConfig()
+						.setNamespace(Objects.toString(properties.getNamespace(),
+								DEFAULT_NAMESPACE))
+						.setGroup(groupFor(uris, properties)).setDataId(dataId)
+						.setSuffix(suffixFor(uris, properties))
+						.setRefreshEnabled(refreshEnabledFor(uris, properties)));
+	}
+
+	private String getUri(ConfigDataLocation location, NacosConfigProperties properties) {
+		String path = location.getNonPrefixedValue(getPrefix());
+		if (StringUtils.isBlank(path)) {
+			path = "/";
+		}
+		if (!path.startsWith("/")) {
+			path = "/" + path;
+		}
+		return properties.getServerAddr() + path;
+	}
+
+	private void registerOrUpdateIndexes(NacosConfigProperties properties,
+										 ConfigurableBootstrapContext bootstrapContext) {
+		String namespace = Objects.toString(properties.getNamespace(), DEFAULT_NAMESPACE);
+		ConfigService configService;
+		try {
+			configService = ConfigFactory.createConfigService(properties.assembleConfigServiceProperties());
+		} catch (NacosException e) {
+			return;
+		}
+		if (bootstrapContext.isRegistered(ConfigServiceIndexes.class)) {
+			ConfigServiceIndexes indexes = bootstrapContext
+					.get(ConfigServiceIndexes.class);
+			indexes.getIndexes().putIfAbsent(namespace, configService);
```

Comment:
In fact, namespace is the only one, `ConfigServiceIndexes `is not very useful?

Reply:
yep, refactored

Reply:
use `NacosConfigManager` create `ConfigService` ?

Reply:
better choice, fixed